### PR TITLE
added support for regex in proxy options

### PR DESCRIPTION
**Pull request by jamon**
provide configurations where the path is a regular expression, eg:

```
        proxy: [
            {
                path: /\/google(.*)/,
                target: "http://www.google.com/"
            }
        ],
```

### CODE REVIEW

File: bin/webpack-dev-server.js [hunk 1]
```diff
@@ -135,17 +135,14 @@ new Server(webpack(wpOpt), options).listen(options.port, options.host, function(
...
-			} else {
-				target = options.proxy[path].target;
-			}
-			console.log('proxying '+path+' to '+target);
+	if (options.proxy && !Array.isArray(options.proxy)) {
+		options.proxy = Object.keys(options.proxy).map(function (path) {
+			return ({
+				path: path,
+				target: typeof options.proxy[path] === "string" ? options.proxy[path] : options.proxy[path].target
+			});
```
Comment (sokra) [inline bin/webpack-dev-server.js]: This part should be moved into `lib/Server.js` to allow this config style without the CLI too.

**Comment by AprilArcus** [comment]
This would be super useful for us! We are working on a mock server that has to support a very heterogenous legacy route scheme, and we'd like to be able to white-list certain routes and proxy the rest.

**Comment by sokra** [comment]
@AprilArcus take a look at the note, fix it and create another PR.

**Comment by nthtran** [comment]
I fixed this in #187. Please a have look.

**Comment by PavelGavlik** [comment]
Shouldn't we close this PR when #187 has superseded this PR and it has been merged?

**Pull request closed without merging**